### PR TITLE
Auth setup

### PR DESCRIPTION
**Pull request by lazedo**
important fixes to kazoo_auth, kazoo_attachments

still needs cb_auth changes in subsequent PR

### CODE REVIEW

**Comment by jamhed** [comment]
@lazedo in order for makebusy to pass this needs rebase (to include new cb_system_configs)

File: core/kazoo_auth/priv/couchdb/providers/google.json [hunk 1]
```diff
@@ -1,9 +1,30 @@
 {
     "_id": "google",
-    "enabled": true,
```
Comment (icehess) [inline core/kazoo_auth/priv/couchdb/providers/google.json]: Why removing this? I added this key here to allow disable or enable a specific provider.

File: core/kazoo_auth/priv/couchdb/views/providers.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "_id": "_design/providers",
```
Comment (icehess) [inline core/kazoo_auth/priv/couchdb/views/providers.json]: Could you move `enabled_providers_by_type` and `providers_by_type` from `auth.json` here and update `cb_multi_factor`?

File: core/kazoo_auth/priv/couchdb/providers/google.json (hunk 1 above)
Comment (lazedo) [inline core/kazoo_auth/priv/couchdb/providers/google.json]: enable where ?

File: core/kazoo_auth/priv/couchdb/views/providers.json (hunk 2 above)
Comment (lazedo) [inline core/kazoo_auth/priv/couchdb/views/providers.json]: why ?

File: core/kazoo_auth/priv/couchdb/providers/google.json (hunk 1 above)
Comment (icehess) [inline core/kazoo_auth/priv/couchdb/providers/google.json]: for multi_factor I added this key, and in `kazoo_auth_mfa` it checks if the module is enabled by account or system.

Also the view `enabled_providers_by_type` is using this key to list enabled provider.

**Pull request merged**